jazkarta/shop/browser/cart.py

```diff
@@ -116,3 +117,21 @@ def format_discount(coupon):
     else:
         discount = '%s%%' % int(coupon.amount)
     return discount
+
+
+class RelatedProductsView(CartViewMixin, BrowserView):
+
+    def products(self):
+        products_in_cart = [el.product for el in self.cart.items]
+        result = []
+
+        for item in products_in_cart:
+            product = IProduct(item)
+            for related in product.related_products:
+                obj = related.to_object
+                if obj not in result and obj not in products_in_cart:
+                    result.append({'obj': obj, 'image_url': None})
+                    with_image = IProductImage(obj, None)
```

Comment:
This adapter seems like overkill. Maybe just a `get_product_data(obj)` that returns a dict of data used by the view for each related object. Specific sites can override the view (using `layer` in the view registration) and implement their own `get_product_data` method and template.